cmd/cluster-controller/main.go

```diff
@@ -91,10 +96,40 @@ func main() {
 	}
 	kcpSharedInformerFactory := kcpexternalversions.NewSharedInformerFactoryWithOptions(kcpclient.NewForConfigOrDie(r), resyncPeriod)
 	crdSharedInformerFactory := crdexternalversions.NewSharedInformerFactoryWithOptions(apiextensionsclient.NewForConfigOrDie(r), resyncPeriod)
-	if err := options.Options.Complete(kubeconfig, kcpSharedInformerFactory, crdSharedInformerFactory).Start(ctx); err != nil {
+	c := options.Options.Complete(kubeconfig, kcpSharedInformerFactory, crdSharedInformerFactory)
+	cluster, apiresource, err := c.New()
+	if err != nil {
 		fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
 
+	kcpSharedInformerFactory.Start(ctx.Done())
```

Comment:
This would appear to largely duplicate the init code in the server. Maybe factor out the common code for reuse?

Reply:
shared informer factories IMO are primary responsibility of the caller, i.e. main.go or some other embedding code like the server. Would not overabstract this. There are only 3 factories anyway.

Reply:
...and any test fixture that wants to start the controller in-process? Is there a good reason not to make the controller easy to start?

Reply:
That asks for some framework-like thing to easily create factories and do the start+waitForSync right, rather than a specific helper.